### PR TITLE
(test) Some miscellaneous test fixes

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR makes some miscellaneous fixes to some tests (including updates to some Zod schema validation logic) and updates some translation strings in the Patient Chart package to match our sentence case convention for strings in the UI.

### CODE REVIEW

File: tools/test-utils.tsx [hunk 1]
```diff
@@ -65,7 +66,7 @@ export const mockPatient = {
       value: '100732HE',
     },
     {
-      id: '1f0ad7a1-430f-4397-b571-59ea654a52db',
+      id: 'ceda35a8-a445-455d-9928-ac088692190a',
```
Comment (denniskigen) [inline tools/test-utils.tsx:69]: Ditto about the reused identifiers.

File: __mocks__/mockDeceasedPatient.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ export const mockDeceasedPatient = {
       value: '100732HE',
     },
     {
-      id: '1f0ad7a1-430f-4397-b571-59ea654a52db',
+      id: 'ceda35a8-a445-455d-9928-ac088692190a',
```
Comment (denniskigen) [inline __mocks__/mockDeceasedPatient.ts:22]: We had two identifiers here reusing the same UUID.

File: packages/esm-patient-orders-app/src/order-cancellation-form/cancel-order-form.component.tsx [hunk 3]
```diff
@@ -46,8 +46,8 @@ const OrderCancellationForm: React.FC<OrderCancellationFormProps> = ({
         .refine((date) => date >= dayjs().startOf('day').toDate(), {
           message: t('dateCannotBeBeforeToday', 'Date cannot be before today'),
         }),
-      reasonForCancellation: z.string({
-        required_error: t('reasonForCancellationRequired', 'Reason for cancellation is required'),
```
Comment (denniskigen) [inline packages/esm-patient-orders-app/src/order-cancellation-form/cancel-order-form.component.tsx:50]: The original check only validates that the field input is not null or undefined. The `min(1)` check explicitly validates that the string is not empty. It's also arguably more readable.

File: packages/esm-patient-notes-app/src/notes/visit-notes-form.workspace.tsx [hunk 4]
```diff
@@ -78,8 +78,8 @@ interface DiagnosisSearchProps {
 
 const visitNoteFormSchema = z.object({
   noteDate: z.date(),
-  primaryDiagnosisSearch: z.string({
-    required_error: 'Choose at least one primary diagnosis',
+  primaryDiagnosisSearch: z.string().min(1, {
+    message: 'Choose at least one primary diagnosis',
   }),
```
Comment (denniskigen) [inline packages/esm-patient-notes-app/src/notes/visit-notes-form.workspace.tsx:83]: Rationale explained below.

File: packages/esm-patient-notes-app/src/notes/visit-notes-form.workspace.tsx [hunk 5]
```diff
@@ -129,6 +129,7 @@ const VisitNotesForm: React.FC<DefaultPatientWorkspaceProps> = ({
     mode: 'onSubmit',
     resolver: zodResolver(visitNoteFormSchema),
     defaultValues: {
+      primaryDiagnosisSearch: '',
```
Comment (denniskigen) [inline packages/esm-patient-notes-app/src/notes/visit-notes-form.workspace.tsx:132]: Explicitly sets a default value so the input is controlled.

File: packages/esm-patient-labs-app/src/lab-orders/add-lab-order/lab-order-form.component.tsx [hunk 6]
```diff
@@ -72,12 +72,9 @@ export function LabOrderForm({
...
           ? z
-              .string({
-                required_error: translateFrom(
-                  moduleName,
-                  'addLabOrderLabOrderReasonRequired',
-                  'Order reason is required',
-                ),
+              .string()
+              .min(1, {
+                message: translateFrom(moduleName, 'addLabOrderLabOrderReasonRequired', 'Order reason is required'),
```
Comment (denniskigen) [inline packages/esm-patient-labs-app/src/lab-orders/add-lab-order/lab-order-form.component.tsx:77]: Rationale explained below.

File: packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx [hunk 7]
```diff
@@ -130,8 +130,9 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
         ...acc,
         [uuid]: required
           ? z
-              .string({
-                required_error: t('fieldRequired', 'This field is required'),
+              .string()
+              .min(1, {
+                message: t('fieldRequired', 'This field is required'),
               })
```
Comment (denniskigen) [inline packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx:136]: Rationale explained below.

**Review by jayasanka-sack — APPROVED** [review]
Thanks @denniskigen !

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]
The visit note E2E test is failing here. I guess it's due to validation rule changes. Could you please check?
Screenshot 2024-07-08 at 12 33 34

**Review by jayasanka-sack — APPROVED** [review]
Thanks @denniskigen !

**Pull request merged**